scripts/prepare_docker_buildinfo.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+IMAGENAME=$1
+DOCKERFILE=$2
+ARCH=$3
+DOCKERFILE_TARGE=$4
+DISTRO=$5
+
+[ -z "$BUILD_SLAVE" ] && BUILD_SLAVE=n
+[ -z "$DOCKERFILE_TARGE" ] && DOCKERFILE_TARGE=$DOCKERFILE
+DOCKERFILE_PATH=$(dirname "$DOCKERFILE_TARGE")
+BUILDINFO_PATH="${DOCKERFILE_PATH}/buildinfo"
+BUILDINFO_VERSION_PATH="${BUILDINFO_PATH}/versions"
+
+[ -d $BUILDINFO_PATH ] && rm -rf $BUILDINFO_PATH
+mkdir -p $BUILDINFO_VERSION_PATH
+
+# Get the debian distribution from the docker base image
+if [ -z "$DISTRO" ]; then
+    DOCKER_BASE_IMAGE=$(grep "^FROM" $DOCKERFILE | head -n 1 | awk '{print $2}')
+    DISTRO=$(docker run --rm --entrypoint "" $DOCKER_BASE_IMAGE cat /etc/os-release | grep VERSION_CODENAME | cut -d= -f2)
+    [ -z "$DISTRO" ] && DISTRO=jessie
+fi
+
+DOCKERFILE_PRE_SCRIPT='# Auto-Generated for buildinfo 
+COPY ["buildinfo", "/usr/local/share/buildinfo"]
+RUN dpkg -i /usr/local/share/buildinfo/sonic-build-tools_1.0_all.deb
+RUN apt-mark hold sonic-build-tools
+RUN pre_run_buildinfo'
```

Comment:
Is it possible to embed these actions into .deb installation script?

Reply:
It is to collect the versions info, may be not good to embed to the installation script.

---
In reply to: [537938558](https://github.com/Azure/sonic-buildimage/pull/5718#discussion_r537938558) [](ancestors = 537938558)